src/io/flutter/utils/FlutterModuleUtils.java

```diff
@@ -74,6 +84,69 @@ public static boolean usesFlutter(@NotNull Project project) {
     return CollectionUtils.anyMatch(getModules(project), FlutterModuleUtils::usesFlutter);
   }
 
+
+  /**
+   * Create a Flutter run configuration.
+   *
+   * @param project    the project
+   * @param main       the target main
+   * @param workingDir the working directory
+   */
+  public static void createRunConfig(@NotNull Project project, @Nullable VirtualFile main, @NotNull VirtualFile workingDir) {
+    final ConfigurationFactory[] factories = FlutterRunConfigurationType.getInstance().getConfigurationFactories();
+    final Optional<ConfigurationFactory> factory =
+      Arrays.stream(factories).filter((f) -> f instanceof FlutterRunConfigurationType.FlutterConfigurationFactory).findFirst();
+    assert (factory.isPresent());
+    final ConfigurationFactory configurationFactory = factory.get();
+
+    final RunManager runManager = RunManager.getInstance(project);
+    final List<RunConfiguration> configurations = runManager.getConfigurationsList(FlutterRunConfigurationType.getInstance());
+
+    // If the target project has no flutter run configurations, create one.
+    if (configurations.isEmpty()) {
+      final RunnerAndConfigurationSettings settings =
+        runManager.createRunConfiguration(project.getName(), configurationFactory);
+      final FlutterRunConfiguration configuration = (FlutterRunConfiguration)settings.getConfiguration();
+
+      // Set config name.
+      String name = configuration.suggestedName();
+      if (name == null) {
+        name = project.getName();
+      }
+      configuration.setName(name);
+
+      // Setup parameters.
+      final FlutterRunnerParameters parameters = configuration.getRunnerParameters();
+
+      // Add main if appropriate.
+      if (main != null && main.exists()) {
+        parameters.setFilePath(main.getPath());
+      }
+
+      parameters.setWorkingDirectory(workingDir.getPath());
+
+      runManager.addConfiguration(settings, false);
+      runManager.setSelectedConfiguration(settings);
+    }
+  }
+
+  /**
+   * Returns a pair describing the main file (`lib/main.dart`) and it's content root for
+   * the given module.  Note that if there is more than one defined main in the given module,
+   * the first will be returned.  If none is found, returns `null`.
+   */
+  @Nullable
+  public static Pair<VirtualFile, VirtualFile> findFlutterMain(@NotNull Module module) {
```

Comment:
Optional: maybe a tiny class would be better, just to avoid confusing the files? (I tend to avoid Pair.)

```
public class FileWithContext {
  public final VirtualFile file;
  public final VirtualFile contextRoot;
  ...
}
```